mailpoet/lib/Doctrine/EventListeners/NewsletterListener.php

```diff
@@ -0,0 +1,50 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Doctrine\EventListeners;
+
+use MailPoet\Entities\NewsletterEntity;
+use MailPoet\Entities\WpPostEntity;
+use MailPoet\WP\Functions as WPFunctions;
+use MailPoetVendor\Doctrine\ORM\Event\LifecycleEventArgs;
+
+class NewsletterListener {
+  /** @var WPFunctions */
+  private $wp;
+
+  public function __construct(
+    WPFunctions $wp
+  ) {
+    $this->wp = $wp;
+  }
+
+  public function prePersist(): void {
```

Comment:
I don't think this is necessary. We could update `EventListenersBaseTest::replaceListeners()` by adding an optional parameter with listener events that should be replaced. When the new parameter is optional with the default value `[Events::prePersist, Events::preUpdate]`, then we don't need to touch the other tests extending this class. What do you think about this approach?